MAINTAINING.md

```diff
@@ -142,49 +142,78 @@ Internal APIs:
 
 ## Managing Issues
 
-### User Impacting Changes
-
-__RULE__: Any commit that may affect the end user in some way MUST have either a
-changelog entry OR MUST have an issue marked with one of the following labels
-OR both.  We can have more than one of these labels on the same issue e.g.
-breaking, enhancement:
-
-* breaking
-* deprecating
-* enhancement
-* performance
-* usability
-* documentation
-* bug
-
-Note that if you are making a big feature change you may have a single issue
-for that feature and attach many commits with it. So you do not necessarily
-need to have an issue for each commit.
-
-### Maintenance Changes
-
-Commits that do not impact the end user in any way are not required to have a
-changelog entry or an issue.  Issues that do not have a corresponding commit
-may be left without a label but preferably should be marked with one of the
-following:
+We label the issues in different dimensions based on the lifecyle and different
+management aspects of the issue. The folowing sections discuss the labels used
+to manage the issues. If a new label is required, it should be discussed before
+creating it.
+
+### Disposition
+
+The level-1 triaging of the issue determines the current disposition
+of the issue. If no change is required the issue must have one of the
+following labels:
 
+* invalid
 * question
 * discussion
-* maintenance
-* testing
-* invalid
+* duplicate
 * wontfix
 
-### Feature Labels
+If a change is required we need to do level-2 triage of the issue, see the
+sections below.
 
-For big features with many issues we can introduce a custom feature label to
-group the issues together.
+### Change type
 
-### Other Labels
+When a change is required we need to put one of the __change type__
+labels as part of level-2 triaging:
 
-* help wanted
-* duplicate
-* deferred
+* performance: User visible impact on performance.
+* usability: It is not convenient to use the library.
+* documentation: documentation is not correct or sufficient.
+* bug: A functionality issue, not working as expected.
+* enhancement: A new feature or enhancement of the product.
+* maintenance: A refactor or any other change with no user visible impact.
+
+In addition we can put a __product aspect__ label describing a feature name or
+any other product specific classification bucket.
+
+* aspect:<aspect name>
+
+### Severity
+
+Optionally we can add a severity label to indicate how severe is the
+impact of the bug/issue, which may determine how important it is to fix it. By
+default the severity is normal, if it is high we put a label:
+
+* sev:high
```

Comment:
I don't see the need for a shorthand,  `severity:high` seems fine.

Reply:
ok.